src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/Serializer/ProtobufOtlpMetricSerializer.cs

```diff
@@ -318,54 +321,12 @@ private static int WriteMetric(byte[] buffer, int writePosition, Metric metric)
 
                     foreach (ref readonly var metricPoint in metric.GetMetricPoints())
                     {
-                        writePosition = ProtobufSerializer.WriteTag(buffer, writePosition, ProtobufOtlpMetricFieldNumberConstants.ExponentialHistogram_Data_Points, ProtobufWireType.LEN);
-                        int dataPointLengthPosition = writePosition;
-                        writePosition += ReserveSizeForLength;
-
-                        var startTime = (ulong)metricPoint.StartTime.ToUnixTimeNanoseconds();
-                        writePosition = ProtobufSerializer.WriteFixed64WithTag(buffer, writePosition, ProtobufOtlpMetricFieldNumberConstants.ExponentialHistogramDataPoint_Start_Time_Unix_Nano, startTime);
-
-                        var endTime = (ulong)metricPoint.EndTime.ToUnixTimeNanoseconds();
-                        writePosition = ProtobufSerializer.WriteFixed64WithTag(buffer, writePosition, ProtobufOtlpMetricFieldNumberConstants.ExponentialHistogramDataPoint_Time_Unix_Nano, endTime);
-
-                        foreach (var tag in metricPoint.Tags)
-                        {
-                            writePosition = WriteTag(buffer, writePosition, tag, ProtobufOtlpMetricFieldNumberConstants.ExponentialHistogramDataPoint_Attributes);
-                        }
-
-                        var sum = metricPoint.GetHistogramSum();
-                        writePosition = ProtobufSerializer.WriteDoubleWithTag(buffer, writePosition, ProtobufOtlpMetricFieldNumberConstants.ExponentialHistogramDataPoint_Sum, sum);
-
-                        var count = (ulong)metricPoint.GetHistogramCount();
-                        writePosition = ProtobufSerializer.WriteFixed64WithTag(buffer, writePosition, ProtobufOtlpMetricFieldNumberConstants.ExponentialHistogramDataPoint_Count, count);
-
-                        if (metricPoint.TryGetHistogramMinMaxValues(out double min, out double max))
-                        {
-                            writePosition = ProtobufSerializer.WriteDoubleWithTag(buffer, writePosition, ProtobufOtlpMetricFieldNumberConstants.ExponentialHistogramDataPoint_Min, min);
-                            writePosition = ProtobufSerializer.WriteDoubleWithTag(buffer, writePosition, ProtobufOtlpMetricFieldNumberConstants.ExponentialHistogramDataPoint_Max, max);
-                        }
-
-                        var exponentialHistogramData = metricPoint.GetExponentialHistogramData();
+                        writePosition = WriteExponentialHistogramDataPoint(buffer, writePosition, in metricPoint);
```

Comment:
There is a lot of refactoring in this file that is unrelated to the proposed changes. Could you please refactor these outside of this current PR? Doing so either before or after this PR merges would be fine.

Reply:
I'm off for the holidays. I'll look into that first week of January.